### PR TITLE
Support virtual wans

### PR DESCRIPTION
**Pull request by sathish-progress**
### Description

Support Virtual WAN(s)

### Check List

- [X] New functionality includes tests
- [X] All tests pass
- [X] `rake lint` passes
- [X] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: docs/resources/azure_virtual_wans.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+| types              | A list of types for all the resources.             | `type`          |
+| locations          | A list of locations for all the resources.         | `location`      |
+| properties         | A list of Properties all the resources.            | `properties`    |
+
+
+<superscript>*</superscript> For information on how to use filter criteria on plural resources refer to [FilterTable usage](https://github.com/inspec/inspec/blob/master/dev-docs/filtertable-usage.md).
+
+## Examples
+
+### Test that There are virtual WANs of standard type
```
Comment (soumyo13) [inline docs/resources/azure_virtual_wans.md:55]: Can we give example as?

describe azure_virtual_wans do
  its('ids') { should include 'example_id' }
end

**Review by soumyo13 — APPROVED** [review]
LGTM

File: docs/resources/azure_virtual_wan.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+| Property                      | Description                                                       |
+|-------------------------------|-------------------------------------------------------------------|
+| id                            | Resource ID.                     |
+| name                          | Resource name.                                         |
+| type                          | Resource type.                                         |
+| etag                          | A unique read-only string that changes whenever the resource is updated.
+| location                      | Resource location.                                                |
+| properties.provisioningState  | The provisioning state of the virtual WAN resource.                 |
+| properties.disableVpnEncryption| Vpn encryption to be disabled or not.                       |
```
Comment (dkumaras) [inline docs/resources/azure_virtual_wan.md:59]: From Line 55, the field

File: docs/resources/azure_virtual_wans.md (hunk 1 above)
Comment (sathish-progress) [reply docs/resources/azure_virtual_wans.md:55]: ehh we have a better example just below .. and also [here](https://github.com/inspec/inspec-azure/pull/417/files/5d35bd3e9f8082ec18cb5e910ec2f94e78088ea7#diff-8441a2edee86ac64cf5d291138cb3ddc9dbdb79ce28ce8617d3d9147b6c730c1R73)

File: docs/resources/azure_virtual_wan.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+## Availability
+
+### Installation
+
+This resource is available in the [InSpec Azure resource pack](https://github.com/inspec/inspec-azure). For an example, `inspec.yml` file and how to set up your Azure credentials, refer to resource pack [README](../../README.md#Service-Principal).
+
+## Syntax
+
+`name` is a required parameter.
```
Comment (IanMadd) [inline docs/resources/azure_virtual_wan.md:24]: @sathish-progress Is `resource_group` also a required parameter?

File: docs/resources/azure_virtual_wan.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+| properties.type               | The type of the Virtual WAN.                                      |
+
+For properties applicable to all resources, such as `type`, `name`, `id`, `properties`, refer to [`azure_generic_resource`](azure_generic_resource.md#properties).
+
+Also, refer to [Azure documentation](https://docs.microsoft.com/en-us/rest/api/virtualwan/virtual-wans/get) for other properties available.
+Any attribute in the response may be accessed with the key names separated by dots (`.`).
+
+## Examples
+
+### Test that the Virtual WAN's Resource Group Encryption not equals 'be_falsey'
```
Comment (IanMadd) [inline docs/resources/azure_virtual_wan.md:68]:
```suggestion
### Test that a Virtual WAN's encryption is not disabled.
```

File: docs/resources/azure_virtual_wans.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+| etags              | A list of etag for all the Resources.              | `etag`          |
+| types              | A list of types for all the resources.             | `type`          |
+| locations          | A list of locations for all the resources.         | `location`      |
+| properties         | A list of Properties all the resources.            | `properties`    |
+
+<superscript>*</superscript> For information on how to use filter criteria on plural resources refer to [FilterTable usage](https://github.com/inspec/inspec/blob/master/dev-docs/filtertable-usage.md).
+
+## Examples
+
+### Test that the Virtual WANs is of Standard Type
```
Comment (IanMadd) [inline docs/resources/azure_virtual_wans.md:48]:
```suggestion
### Test that the Virtual WAN type is `Standard`
```

File: docs/resources/azure_virtual_wan.md (hunk 3 above)
Comment (sathish-progress) [reply docs/resources/azure_virtual_wan.md:24]: yeah, I will add them ..

Comment (IanMadd) [inline docs/resources/azure_virtual_wan.md:24]:
```suggestion
`name` and `resource_group` are required parameters.
```

**Review by dkumaras — APPROVED** [review]

**Pull request merged**